packages/shared/src/hooks/usePostById.ts

```diff
@@ -22,24 +24,31 @@ const usePostById = ({
   options = {},
 }: UsePostByIdProps): UsePostById => {
   const { tokenRefreshed } = useAuthContext();
+  const isEnabled = !!id && tokenRefreshed;
   const {
     data: postById,
     isLoading,
     isFetched,
   } = useQuery<PostData>(
     ['post', id],
     () => request(graphqlUrl, POST_BY_ID_QUERY, { id }),
-    { ...options, enabled: !!id && tokenRefreshed },
+    {
+      ...options,
+      enabled: isNullOrUndefined(options.enabled)
+        ? isEnabled
+        : isEnabled && options.enabled,
+    },
   );
 
   const post = postById || (options?.initialData as PostData);
 
   return useMemo(
     () => ({
       post: post?.post,
-      isLoading: isLoading || !isFetched || isFetchingNextPage,
+      isFetched,
+      isLoading: isLoading || isFetchingNextPage,
     }),
-    [postById, isLoading],
+    [id, postById, isLoading, options, isFetched, isFetchingNextPage],
```

Comment:
Why do we need all of that in the deps list?

Reply:
I think they are necessary as those values are changing, and we decide some things based on them. This is to keep the values accurate.

Reply:
How come? I see inside the use memo you use only `post, isFetched, isLoading, isFetchingNextPage`

Reply:
Yes, but the `post` is based on the `options`, for when it changes, the value of `post` might change too.